lib/create-contentful-api.ts

```diff
@@ -279,352 +306,360 @@ interface GetConfig {
  * @prop {Function} getGlobalOptions - Link resolver preconfigured with global setting
  * @return {ClientAPI}
  */
-export default function createContentfulApi ({ http, getGlobalOptions }): ContentfulClientApi {
-  const notFoundError = (id) => {
-    const error = new Error('The resource could not be found.')
-    /*
-      error.sys = {
-      type: 'Error',
-      id: 'NotFound'
+
+class NotFoundError extends Error {
+    public readonly sys: { id: string; type: string };
+    public readonly details: { environment: string; id: string; type: string; space: any };
+
+    constructor(id: string, environment: string, space: string) {
+        super('The resource could not be found.');
+        this.sys = {
+            type: 'Error',
+            id: 'NotFound'
+        }
+        this.details = {
+            type: 'Entry',
+            id,
+            environment,
+            space
+        }
+    }
+}
+
+export default function createContentfulApi({http, getGlobalOptions}): ContentfulClientApi {
+    const notFoundError = (id) => {
+        return new NotFoundError(id, getGlobalOptions().environment, getGlobalOptions().space)
```

Comment:
Maybe we should call the `getGlobalOptions` function before and ensure that this is all the same everywhere?  
The mix of the `switchToSpace` side effects and calling the suspiciously named `getGlobalOptions` function gives me a bit of anxiety 😉